src/app/core/session/auth/keycloak/auth-interceptor.service.ts

```diff
@@ -4,10 +4,22 @@ import {
   HttpHandler,
   HttpEvent,
   HttpInterceptor,
+  HttpContextToken,
 } from "@angular/common/http";
 import { Observable } from "rxjs";
 import { AuthService } from "../auth.service";
 
+/**
+ * This context can be used to prevent the Bearer token to be set by this interceptor.
+ * This can be useful when a request goes to a 3rd party service e.g. GitHub.
+ *
+ * Usage:
+ * ```javascript
+ * this.httpClient.get(`...`, { context: new HttpContext().set(AUTH_ENABLED, false) });
+ * ```
+ */
+export const AUTH_ENABLED = new HttpContextToken(() => true);
```

Comment:
From a big picture perspective, wouldn't it be better that developers don't have to actively disable something they are not even aware of being added? (i.e. default auth = false + maybe switching the default to true for requests to any subdomains of the own host)

I can live with this also but probably we will always release a broken version before thinking of putting the parameter in here?

Reply:
Let's see if we start using more external APIs which do not require our auth. If this will be the case then we can look for a more suitable solution.